File: vhdl_toolkit/vhdlConvertor/hdlObjects/CompInstance.java
package hdlObjects;

import java.util.List;
import java.util.Vector;

import org.json.JSONException;
import org.json.JSONObject;

public class CompInstance implements iJsonable {
	public Expr entityName;
	public String name;
	public List<Expr> genericMap;
	public List<Expr> portMap;

	public CompInstance(Expr _entityName) {
		entityName = _entityName;
		genericMap = new Vector<Expr>();
		portMap = new Vector<Expr>();
	}
	@Override
	public JSONObject toJson() throws JSONException {
		JSONObject o = new JSONObject();
		o.put("name", name);
		o.put("entityName", entityName.toJson());
		return o;
	}

}

File: hdl_toolkit/parser/hdlConvertor/src/hdlObjects/compInstance.cpp
#include "compInstance.h"

CompInstance::CompInstance(const char * name, Expr * _entityName) {
	entityName = _entityName;
	this->name = name;
}
PyObject * CompInstance::toJson() const {
	PyObject * d = PyDict_New();
	PyDict_SetItemString(d, "entityName", entityName->toJson());
	return d;
}
